### PR TITLE
Fix e2e tests for another localhub

### PR DESCRIPTION
**Pull request by MMMsc**
1. add e2e_tests_parameter module on hoh-client_test
2. fix the other e2e tests for 2 localhubs & 1 managed cluster
fixed: https://github.com/stolostron/multicluster-global-hub/issues/317

### CODE REVIEW

File: test/pkg/e2e/hoh-application_test.go [hunk 1]
```diff
@@ -29,23 +30,31 @@ const (
...
 var _ = Describe("Deploy the application to the managed cluster", Label("e2e-tests-app"), Ordered, func() {
-	var managedClusterName1 string
-	var managedClusterName2 string
-	var managedClusterUID1 string
-	var managedClusterUID2 string
+	type managedClusterInfo struct {
+		name string
+		uid  string
+	}
+	var managedClusterInfos []managedClusterInfo
```
Comment (yanmxa) [inline test/pkg/e2e/hoh-application_test.go:37]: using the original `managedcluster` struct

File: test/pkg/e2e/hoh-application_test.go [hunk 2]
```diff
@@ -69,7 +78,7 @@ var _ = Describe("Deploy the application to the managed cluster", Label("e2e-tes
 		}
 
 		Eventually(func() error {
-			err := updateClusterLabel(httpClient, patches, httpToken, managedClusterUID1)
```
Comment (yanmxa) [inline test/pkg/e2e/hoh-application_test.go:72]: combine line 81 ~ 100 to one eventually.

File: test/pkg/e2e/hoh-label_test.go [hunk 3]
```diff
@@ -45,7 +60,7 @@ var _ = Describe("Updating cluster label from HoH manager", Label("e2e-tests-lab
 		}
 
 		Eventually(func() error {
-			err := updateClusterLabel(httpClient, patches, httpToken, managedClusterUID)
```
Comment (yanmxa) [inline test/pkg/e2e/hoh-label_test.go:48]: Integrate these two eventually to one.

File: test/pkg/e2e/hoh-label_test.go [hunk 4]
```diff
@@ -67,36 +82,72 @@ var _ = Describe("Updating cluster label from HoH manager", Label("e2e-tests-lab
...
-				Op:    "remove",
-				Path:  "/metadata/labels/" + CLUSTER_LABEL_KEY,
-				Value: CLUSTER_LABEL_VALUE,
-			},
-		}
-		Eventually(func() error {
-			err := updateClusterLabel(httpClient, patches, httpToken, managedClusterUID)
-			if err != nil {
-				return err
+	It("add the label to the managed cluster", func() {
```
Comment (yanmxa) [inline test/pkg/e2e/hoh-label_test.go:85]: as above.

File: test/pkg/e2e/hoh-label_test.go [hunk 5]
```diff
@@ -67,36 +82,72 @@ var _ = Describe("Updating cluster label from HoH manager", Label("e2e-tests-lab
...
+				{
+					Op:    "remove",
+					Path:  "/metadata/labels/" + CLUSTER_LABEL_KEY,
+					Value: CLUSTER_LABEL_VALUE,
+				},
 			}
-			return nil
-		}, 3*time.Minute, 5*time.Second).ShouldNot(HaveOccurred())
+			Eventually(func() error {
+				err := updateClusterLabel(httpClient, patches, httpToken, managedClusterInfo.uid)
```
Comment (yanmxa) [inline test/pkg/e2e/hoh-label_test.go:129]: as above.

File: test/pkg/e2e/hoh-local-policy_test.go [hunk 6]
```diff
@@ -221,37 +253,46 @@ var _ = Describe("Apply local policy to the managed clusters", Ordered,
...
+						}
+						for _, finalizer := range policy.Finalizers {
+							if finalizer == constants.GlobalHubCleanupFinalizer {
+								foundCount += 1
+								break
+							}
 						}
 					}
-					return nil
+					if foundCount == len(leafhubClients) {
```
Comment (yanmxa) [inline test/pkg/e2e/hoh-local-policy_test.go:273]: It's not right here, we should check that the finalizer should not add to the resource, but it seems like you did the opposite.

File: test/pkg/e2e/hoh-local-policy_test.go [hunk 7]
```diff
@@ -260,18 +296,25 @@ var _ = Describe("Apply local policy to the managed clusters", Ordered,
...
+							return err
 						}
+						for _, finalizer := range placementrule.Finalizers {
+							if finalizer == constants.GlobalHubCleanupFinalizer {
+								foundCount += 1
+								break
+							}
+						}
+					}
+					if foundCount != len(leafhubClients) {
```
Comment (yanmxa) [inline test/pkg/e2e/hoh-local-policy_test.go:316]: The expected placementrule shouldn't add the finalizer, same as local policy.

**Comment by MMMsc** [comment]
@yanmxa

**Pull request closed without merging**